hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/EventServiceImpl.java

```diff
@@ -441,16 +492,10 @@ public void executeEventCallback(Runnable callback) {
 
     @Override
     public void handle(Packet packet) {
-        try {
-            eventExecutor.execute(new RemoteEventProcessor(this, packet));
-        } catch (RejectedExecutionException e) {
-            rejectedCount.inc();
-
-            if (eventExecutor.isLive()) {
-                Connection conn = packet.getConn();
-                String endpoint = conn.getEndPoint() != null ? conn.getEndPoint().toString() : conn.toString();
-                logFailure("EventQueue overloaded! Failed to process event packet sent from: %s", endpoint);
-            }
+        EventEnvelope eventEnvelope = (EventEnvelope) nodeEngine.toObject(packet);
```

Comment:
yes, just did a quick test too. No collisions on TopicEvent, I agree. But I seen the confusing output many times when objects created at the same time (with default toString() impl) has the same output.
The output from my test is:

`
time taken for TopicEvent: 405; collisions: 0
time taken for AtomicLong: 65; collisions: 0
time taken for currentTimeMillis: 52; collisions: 999948
time taken for nanoTime: 126; collisions: 752389
`

the number if iterations is 1 million. So, I'd use AtomicLong as it is a bit faster. Do you agree?